frontend/src/shared/helpers.tsx

```diff
@@ -467,23 +466,28 @@ const toolTipIcon =
 function setupEditableSampleFields(samplesColDefs: ColDef[]) {
   samplesColDefs.forEach((colDef) => {
     colDef.cellClassRules = {
-      unsubmittedChange: (params: any) => {
-        const changes = params.context.getChanges();
-        const changedValue = changes?.find((change: any) => {
+      unsubmittedChange: (params: CellClassParams) => {
+        const changes: SampleChange[] = params.context.getChanges();
+        const changedValue = changes?.find((change) => {
           return (
             change.fieldName === params.colDef.field &&
             change.primaryId === params.data.primaryId
           );
         });
         return changedValue !== undefined;
       },
+      "costCenter-validation-error": (params: CellClassParams) => {
+        if (params.colDef.field === "costCenter") {
+          return !isValidCostCenter(params.value);
+        }
+        return false;
+      },
```

Comment:
Conditionally activates the `costCenter-validation-error` CSS class if the Cost Center input value is invalid.